Userland/Libraries/LibGfx/ImageFormats/TIFFLoader.cpp

```diff
@@ -278,40 +278,45 @@ class TIFFLoadingContext {
 
         Variant<ExifOrientedBitmap, ExifOrientedCMYKBitmap> oriented_bitmap = TRY(([&]() -> ErrorOr<Variant<ExifOrientedBitmap, ExifOrientedCMYKBitmap>> {
             if (metadata().photometric_interpretation() == PhotometricInterpretation::CMYK)
                 return ExifOrientedCMYKBitmap::create(*metadata().orientation(), { *metadata().image_width(), *metadata().image_length() });
             return ExifOrientedBitmap::create(*metadata().orientation(), { *metadata().image_width(), *metadata().image_length() }, BitmapFormat::BGRA8888);
         }()));
 
-        for (u32 strip_index = 0; strip_index < strips_offset.size(); ++strip_index) {
-            TRY(m_stream->seek(strips_offset[strip_index]));
-
-            auto const strip_width = *m_metadata.image_width();
-            auto const rows_in_strip = strip_index < strips_offset.size() - 1 ? rows_per_strip : *m_metadata.image_length() - rows_per_strip * strip_index;
+        for (u32 segment_index = 0; segment_index < offsets.size(); ++segment_index) {
+            TRY(m_stream->seek(offsets[segment_index]));
 
-            auto const decoded_bytes = TRY(strip_decoder(strip_byte_counts[strip_index], { strip_width, rows_in_strip }));
-            auto decoded_strip = make<FixedMemoryStream>(decoded_bytes);
-            auto decoded_stream = make<BigEndianInputBitStream>(move(decoded_strip));
+            auto const rows_in_segment = segment_index < offsets.size() - 1 ? segment_length : *m_metadata.image_length() - segment_length * segment_index;
+            auto const decoded_bytes = TRY(segment_decoder(byte_counts[segment_index], { segment_width, rows_in_segment }));
+            auto decoded_segment = make<FixedMemoryStream>(decoded_bytes);
+            auto decoded_stream = make<BigEndianInputBitStream>(move(decoded_segment));
 
-            for (u32 row = 0; row < rows_per_strip; row++) {
-                auto const absolute_row = row + rows_per_strip * strip_index;
+            for (u32 row = 0; row < segment_length; row++) {
+                auto const absolute_row = row + segment_length * (segment_index / segment_per_rows);
                 if (absolute_row >= *m_metadata.image_length())
                     break;
 
                 Optional<Color> last_color {};
 
-                for (u32 column = 0; column < *m_metadata.image_width(); ++column) {
+                for (u32 column = 0; column < segment_width; ++column) {
+                    auto const absolute_column = column + segment_width * (segment_index % segment_per_rows);
+
                     if (metadata().photometric_interpretation() == PhotometricInterpretation::CMYK) {
                         auto const cmyk = TRY(read_color_cmyk(*decoded_stream));
-                        oriented_bitmap.get<ExifOrientedCMYKBitmap>().set_pixel(column, absolute_row, cmyk);
+                        if (absolute_column >= *m_metadata.image_width())
```

Comment:
does this happen in practice? when? (maybe worth a comment?)

Reply:
Yeah, quite often actually. Tiles are always padded, and this line allows us to skip this padding.